zsl/resource/model_resource.py

```diff
@@ -83,11 +88,18 @@ class ResourceQueryContext(object):
     """
 
     def __init__(self, params, args, data):
-        self._params = params
+        # type: (dict, list, dict) -> ()
         self._args = args.copy()
         self._args_original = args
         self._data = data
 
+        # test if params is a list
+        try:
+            len(params)
```

Comment:
We'll extract an issue, where we can discuss.

It depends on usage, we expect list/tuple and need the index or we just need iterable? We can make a simple test and if something more complex arises, then we can improve. Anyways we should use the extracted simple method everywhere across "parsing" so that we are consitent.